tests/CompileTest.hs

```diff
@@ -45,22 +46,25 @@ withinRange'CH :: Version -> VersionRange -> Bool
 withinRange'CH v r =
     withinRange (fromDataVersion v) r
 
-setupHOME :: IO ()
-setupHOME = do
-  mhome <- lookupEnv "HOME"
-  case mhome of
-    Just home -> do
-      exists <- doesDirectoryExist home
-      when (not exists) createHOME
-    Nothing -> createHOME
-
-createHOME :: IO ()
-createHOME = do
-  tmp <- fromMaybe "/tmp" <$> lookupEnv "TMPDIR"
-  let home = tmp </> "compile-test-home"
-  _ <- rawSystem "rm" ["-r", home]
-  createDirectory home
-  setEnv "HOME" home
+setupTestEnv :: IO ()
+setupTestEnv = do
+  eHome <- tryIOError getHomeDirectory
+  exists <- case eHome of
+              Right dir -> doesDirectoryExist dir
+              Left _ -> return False
+  -- If HOME exists we assume that default CABAL_DIR
+  -- and XDG_HOME will exist too
+  when (not exists) createDirs
+  where
+    createDirs = do
+      tmp <- getTemporaryDirectory
+      let cabalDir = tmp </> "compile-test-cabal-dir"
+      let xdgHome  = tmp </> "compile-test-xdg-home"
+      forM [cabalDir, xdgHome] $ \d -> do
+        removeDirectoryRecursive d
+        createDirectory d
+      setEnv "CABAL_DIR" cabalDir
+      setEnv "XDG_HOME" xdgHome
```

Comment:
It's just called `HOME`. Not sure where you got the idea that `XDG_HOME` is a thing?

Reply:
Ah, I think you mixed that up with `XDG_CACHE_HOME` which is the equivalent of `~/.cache` whereas `HOME` is `~/`. Like I said earlier you should create a directory for both.